ENGAGEHF/Dashboard/Messages/MessageRow.swift

```diff
@@ -60,54 +57,95 @@ struct MessageRow: View {
             .accessibilityLabel(message.action.localizedDescription.localizedString() + " Symbol")
     }
     
+    private var processingStateView: some View {
+        HStack(spacing: 8) {
+            ProgressView()
+                .controlSize(.small)
+            Text(processingStateText)
+                .font(.caption)
+                .foregroundStyle(.secondary)
+        }
+        .padding(.vertical, 4)
+        .padding(.horizontal, 8)
+        .background {
+            Capsule()
+                .fill(.secondary.opacity(0.1))
+        }
+    }
+    
+    private var processingStateText: String {
+        if let processingState = message.processingState {
+            switch processingState.type {
+            case .healthMeasurement(let count):
+                return "Processing \(count) measurement\(count == 1 ? "" : "s")..."
+            case .questionnaire:
+                return "Processing questionnaire..."
+            }
+        }
+        return "Processing..."
```

Comment:
Please ensure that these strings are localized. I would suggest to move this string generation in the processing state type and localize them there, we already have some similar code there.